index.js

```diff
@@ -69,3 +69,16 @@ browser.webRequest.onHeadersReceived.addListener(
     { urls: ['<all_urls>'] },
     ['blocking', 'responseHeaders']
 );
+
+browser.runtime.onMessage.addListener(function (message, sender, sendResponse) {
+    if (message.type === 'ONION_LOCATION_FOUND') {
```

Comment:
This could be exploited by 3rd party JS running on a site. Is there any way to avoid that?